### ISSUE TITLE
is the project still alive ?

### ISSUE THREAD

**Issue by eoli3n** [elkowar/eww#970]
### Description of the requested feature

Latest release is more than one year ago : 4 Sept 2022, and latest commit is 3 months ago.
This project seems great but misses a lot of basic features, and default widgets.
It deserves higher commit rate.

**Comment by aacebedo** [comment]
That would be very sad because the project is very promising as a great contender in the status bar environment.

Is there anything that could be done in order to help? For instance on merging some PRs ?

**Comment by eoli3n** [comment]
Yes, and people seems very excited about it. If @elkowar doesn't have time to manage this, maybe he could find someone who is, to give him extra keys. The last thing a project need is a fork.

**Comment by aacebedo** [comment]
> Yes, people seems very excited about it. If @elkowar doesn't have time to manage this, maybe he could find someone who has, to give him extra keys. The last thing a project need is a fork.

100% Agreed.

**Comment by eoli3n** [comment]
~40 PRs are waiting for review.
And people are asking for a release and project cleaning
https://github.com/elkowar/eww/issues/963
https://github.com/elkowar/eww/issues/907
https://github.com/elkowar/eww/issues/929
https://github.com/elkowar/eww/issues/937
https://github.com/elkowar/eww/issues/905

Basic features are missing like, ability to use for loop at top level to manage multihead setup
https://github.com/elkowar/eww/issues/935
https://github.com/elkowar/eww/issues/875
And some default scripts/widget, for exemple for sway workspaces
https://github.com/elkowar/eww/issues/917

**Comment by Animeshz** [comment]
If anybody want to directly contact / ping him, he's actually really active on [r/unixporn discord server](https://discord.gg/TnJ4h5K)!

**Comment by awsomearvinder** [comment]
Looking at discord and viewing his message history it doesn't sound like he's not doing eww stuff anymore, but rather just that he hasn't had time to work on it recently. I'd post a screenshot of what he said but it seems a bit off to me to post, since it's not an actual official statement from him...

**Comment by eoli3n** [comment]
Chasing and quoting him seems not the right thing to do. Wait and see when he will answer here.

**Comment by vandrw** [comment]
As you've mentioned, I believe that we could greatly expand this project with default widgets, and perhaps a place where users could share them. For example, many dotfiles I've come across have had various scripts written from scratch to visualize battery levels, check CPU temperature, etc. This can be hard for someone new to work on, and could be a great way to gain users to test the code further.

I will become more available in the next 2 months, and I could potentially help if we were to fork the project and make a community maintained version (similar to eza). However, how many of us would be willing to take on such a task?

**Comment by aacebedo** [comment]
> As you've mentioned, I believe that we could greatly expand this project with default widgets, and perhaps a place where users could share them. For example, many dotfiles I've come across have had various scripts written from scratch to visualize battery levels, check CPU temperature, etc. This can be hard for someone new to work on, and could be a great way to gain users to test the code further.
> 
> If no word comes from the maintainer, I will become more available in the next 2 months, and I could potentially help if we were to fork the project and make a community maintained version (similar to eza). However, how many of us would be willing to take on such a task?

I'll be glad to help if needed.

**Comment by eoli3n** [comment]
> As you've mentioned, I believe that we could greatly expand this project with default widgets, and perhaps a place where users could share them. For example, many dotfiles I've come across have had various scripts written from scratch to visualize battery levels, check CPU temperature, etc. This can be hard for someone new to work on, and could be a great way to gain users to test the code further.

Hard-coded widgets, like Waybar provides them, seems better to me. About community shares, we could use the eww wiki, but maintainer needs to open RW for everyone, at least on a specific section.

> If no word comes from the maintainer, I will become more available in the next 2 months, and I could potentially help if we were to fork the project and make a community maintained version (similar to eza). However, how many of us would be willing to take on such a task?

My belief is that forces should not be splitted for now, nothing should be done before his statement. Contribute in the main way, with PR to the actual maintainer, and see what he does with it.
I guess that if you are ready to do things in the way the main maintainer could accept it, maybe it could lead to extra keys for you on the project.

@elkowar would you like to recruit a co-maintainer ?

**Comment by elkowar** [comment]
Okay, let's actually make a statement here:
Eww is definitely not dead, and it's true that part of the issue is that I've just not had all that much time to put into eww lately.
I'd generally be very happy to find a viable co-maintainer! However, i'd love for them to be someone I'm actually somewhat in contact with personally, of course.

I know this doesn't necessarily promise much, but I _am_ planning to get back into working on eww more in the not-so-distant future. There's a couple blockers for some of the "basic" functionality that's being requested, architecturally, however there are some other PRs that I fully agree have been stuck for way too long -- the most obvious one being the window-arguments PR. I'd also _love_ to get systray actually into eww!

**Comment by elkowar** [comment]
About default widgets: I'm very unsure about providing that sort of thing, as I'd think that having "default" full-on widgets as part of eww would basically just result in _less_ creativity in the community. Big part of why I wanted to make eww was to spawn more creativity in the ricing scene, enabling more people to create fully custom widgets -- providing full-on premade widgets as part of eww sounds like it'd just reduce the motivation.

I of course think that sharing user-defined widgets is important. I think however that GitHub is a perfectly valid place to do that. Maybe having an `(import "<github-repo>")` kind of approach could work here?

Another goal of mine is to provide an `ewwtils` tool that would contain logic for things such as WM/compositor specific workspace data fetching, and other more advanced system information that I don't really want to include as part of the built-in magic vars. That way you could just 
`(deflisten workspaces "ewwtils workspaces")`, and ewwtils could be a tool with way more and easier community contribution.

**Comment by aacebedo** [comment]
> About default widgets: I'm very unsure about providing that sort of thing, as I'd think that having "default" full-on widgets as part of eww would basically just result in _less_ creativity in the community. Big part of why I wanted to make eww was to spawn more creativity in the ricing scene, enabling more people to create fully custom widgets -- providing full-on premade widgets as part of eww spounds like it'd just reduce the motivation.
> 
> I of course think that sharing user-defined widgets is important. I think however that GitHub is a perfectly valid place to do that. Maybe having an `(import "<github-repo>")` kind of approach could work here?
> 
> Another goal of mine is to provide an `ewwtils` tool that would contain logic for things such as WM/compositor specific workspace data fetching, and other more advanced system information that I don't really want to include as part of the built-in magic vars. That way you could just `(deflisten workspaces "ewwtils workspaces")`, and ewwtils could be a tool with way more and easier community contribution.

Very happy to see that statement ! 
About the `ewwtils` in fact I had the exact same idea after seeing `eww` using `json` (I think it is a great idea) so I started writing a tool named `machinify` in rust that do exactly what you describe. I'll be happy to share it with and to get feedback if you want.
For the moment it supports hyprland workspaces, active window, PA sound, Power Supply and brightness. Planning to add sway workspace and active window, RAM and CPU monitoring I try as much as possible to keep this pure rust.

**Comment by eoli3n** [comment]
Really happy here too, the wait worthed it :)
